src/PostgREST/Error.hs

```diff
@@ -644,7 +644,7 @@ data Error
   deriving Show
 
 data JwtError
-  = JwtDecodeError Text
+  = JwtDecodeError Text (Maybe Text)
```

Comment:
Maybe we can add a new constructor instead of parametrizing this one? As agreed on https://github.com/PostgREST/postgrest/pull/4101#discussion_r2111599677?

It could make this PR simpler, it affects many lines currently.